### ISSUE TITLE
Replaces messages that contain content other than GitHub links

### ISSUE THREAD

**Issue by kevin-brown** [shu8/SEChat-githubOneboxer#3]
Earlier I [posted a message](https://chat.meta.stackexchange.com/transcript/message/3839576#3839576) in chat and was later informed that this script was messing with it.

Here is the content of the message (markdown)

``` markdown
:3839521 I've taken a shot at building [regexes for Smokey](https://github.com/Charcoal-SE/SmokeDetector/issues/86) based on the domains pulled from the IPs, so hopefully (assuming someone can add them to the code) those will get flagged in the future
```

and rendered

> I've taken a shot at building [regexes for Smokey](https://github.com/Charcoal-SE/SmokeDetector/issues/86) based on the domains pulled from the IPs, so hopefully (assuming someone can add them to the code) those will get flagged in the future

But apparently my GitHub link was the only thing visible

![My message, completely replaced](https://i.stack.imgur.com/qOuRT.png)

---

This appears to be caused by the fact that [the script only checks for a GitHub link](https://github.com/shu8/SEChat-githubOneboxer/blob/ec75bf7d7ee3cad0764287c513a1fd9386ce31f9/githubOneboxer.user.js#L166) and then [completely overwrites the message](https://github.com/shu8/SEChat-githubOneboxer/blob/ec75bf7d7ee3cad0764287c513a1fd9386ce31f9/githubOneboxer.user.js#L86-L96) once it gets the information.

---

This is related to #2, which was posted while I was typing this.

**Comment by Unihedro** [comment]
+1

**Comment by shu8** [comment]
whoops! Can't believe I didn't catch this! I'll change this to only onebox if the link is _the only thing in the message_, just like is done with SE's oneboxes.

(You're completely right about _why_ it happens Kevin.)

I think what I'll do is `trim()` the contents of the message, and then check for a ` ` (space) - that will let me see if there is another word... unless there's a better way any of you know?

**Comment by shu8** [comment]
fixed in 1.1.1

**Comment by shu8** [comment]
@kevin-brown I've also made it so that the action buttons on the left aren't removed - so you can 'reply' to messages and such :)

You can see the change in the above commit^^ :)
